src/components/Login/Login.tsx

```diff
@@ -0,0 +1,163 @@
+    }
+  };
+  return (
+    <Dialog>
+      <DialogTrigger>Sign in / Sign Up</DialogTrigger>
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>
+            {submitType === "sign-in" ? "Sign In" : "Sign Up"}
+          </DialogTitle>
+          <DialogDescription>
+            {submitType === "sign-in"
+              ? "Login into existing account"
+              : "Create a new account"}
+          </DialogDescription>
+        </DialogHeader>
+        <form
+          className="flex flex-col items-center justify-center gap-8"
+          method="post"
+          action="/api/auth/callback/credentials"
+          onSubmit={handleSubmit(onSubmit)}
+        >
+          <p className="h-2 text-center">{formError}</p>
+          <input ref={csrfInput} name="csrfToken" type="hidden" />
+          <Label className="flex flex-col gap-2">
+            Email
+            <Input
+              className="h-10 w-80 bg-slate-200 p-2"
+              type="email"
+              placeholder="Email"
+              {...register("email", {
+                required: "Please, input Email",
+              })}
+            />
+            <span className="h-1">
+              {errors.email ? errors.email.message : ""}
+            </span>
+          </Label>
+          <Label className="flex flex-col gap-2">
+            Password
+            <Input
+              className="h-10 w-80 bg-slate-200 p-2"
+              type="password"
+              placeholder="Password"
+              {...register("password", {
+                required: "Please, input password",
+                validate: (value) => {
+                  if (submitType === "sign-in") return;
+                  if (value.length < 8)
+                    return "Password should be at least 8 characters long";
+                  if (!value.match(/(?=.*?[0-9])/))
+                    return "Password requires at least one digit";
+                  if (
+                    !value.match(/(?=.*[a-z])/) &&
+                    !value.match(/(?=.*[A-Z])/)
+                  )
+                    return "Password requires at least one letter";
+                  if (!value.match(/(?=.*[!@#$%^&*])/))
+                    return "Password requires at least one special character";
+                },
+              })}
+            />
+            <span className="h-1">
+              {errors.password ? errors.password.message : ""}
+            </span>
+          </Label>
+          <span
+            className={
+              "-mt8 text-center text-xs" +
+              (wrongPassCount < 3 ? "opacity-0" : "opacity-100")
+            }
+          >
+            Forgot password?&nbsp;
+            <span className=" cursor-pointer font-semibold  underline">
+              You can resotre it via email
+            </span>
+          </span>
+          <DialogFooter className="-mt-4 flex items-center gap-10 ">
+            <Button className="h-10 w-40" type="submit">
```

Comment:
fixed width? bruh...

Reply:
Why not? i thought it's okay for such components like buttons or inputs?